### PR TITLE
Use `ClockOps` in `ResponseTiming` Middleware

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

File: server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala [hunk 1]
```diff
@@ -43,12 +44,9 @@ object ResponseTiming {
       headerName: CIString = ci"X-Response-Time",
   )(implicit F: Sync[F], clock: Clock[F]): HttpApp[F] =
```
Comment (armanbilge) [inline server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala:45]: `Sync` extends `Clock`, so including both is ambiguous.

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala:45]: I can do this refactoring too 👍🏻

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala:45]: But wait. What's about MiMa?

Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala:45]: Come on, you know the drill 😛 

```scala
@deprecated
def apply[F[_]](
  http: HttpApp[F],
  timeUnit: TimeUnit = MILLISECONDS,
  headerName: CIString = ci"X-Response-Time",
  F: Sync[F],
  clock: Clock[F]
) = ???

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala:45]: mehhhhh!

Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala:45]: Actually, why do we even need `Sync` here?

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala]: Functor + Clock should work well if you wanna get purity at max 😄

Comment (danicheg) [reply server/shared/src/main/scala/org/http4s/server/middleware/ResponseTiming.scala]: Have we beaten all the dragons?

**Review by armanbilge — APPROVED** [review]
Thanks, huge difference!

**Pull request merged**